roadmap/implementers-guide/src/node/backing/prospective-parachains.md

```diff
@@ -0,0 +1,163 @@
+# Prospective Parachains
+
+## Overview
+
+**Purpose:** Tracks and handles prospective parachain fragments and informs
+other backing-stage subsystems of work to be done.
+
+"prospective":
+- [*prə'spɛktɪv*] adj.
+- future, likely, potential
+
+Asynchronous backing changes the runtime to accept parachain candidates from a
+certain allowed range of historic relay-parents. This means we can now build
+*prospective parachains* – that is, trees of potential (but likely) future
+parachain blocks. This is the subsystem responsible for doing so.
+
+Other subsystems such as Backing rely on Prospective Parachains, e.g. for
+determining if a candidate can be seconded. This subsystem is the main
+coordinator of work within the node for the collation and backing phases of
+parachain consensus.
+
+Prospective Parachains is primarily an implementation of fragment trees. It also
+handles concerns such as:
+
+- the relay-chain being forkful
+- session changes
+- predicting validator group assignments
+
+See the following sections for more details.
+
+TODO: move these sections the mod docs if there are too many implementation
+details?
+
+### Fragment Trees
+
+This subsystem builds up fragment trees, which are trees of prospective para
+candidates. Each path through the tree represents a possible state transition
+path for the para. Each potential candidate is a fragment, or a node, in the
+tree. Candidates are validated against constraints as they are added.
+
+This subsystem builds up trees for each relay-chain block in the view, for each
+para. These fragment trees are used for:
+
+- providing backable candidates to other subsystems
+- sanity-checking that candidates can be seconded
+- getting seconded candidates under active leaves
+- etc.
+
+For example, here is a tree with several possible paths:
+
+```
+Para Head registered by the relay chain:     committed_head
```

Comment:
Is `committed_head` the best term here? If we only base fragment trees on included candidates then perhaps `included_head` is more clear.
```suggestion
Para Head registered by the relay chain:     included_head
```